internal/controller/replication.storage/status.go

```diff
@@ -46,26 +49,30 @@ func setPromotedCondition(conditions *[]metav1.Condition, observedGeneration int
 }
 
 // sets conditions when volume promotion was failed.
-func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64) {
+func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64, err string) {
```

Comment:
err is not a good name for an error message, this is the common name for objects implementing the error interface. Maybe use `errorMesssage`?